### ISSUE TITLE
Add `wezterm.action.CopyMode { SetSelectionMode = 'SemanticZone' }` as well as relevant copy and move methods

### ISSUE THREAD

**Issue by ymotongpoo** [wez/wezterm#2346]
**Is your feature request related to a problem? Please describe.**
I would like to have something like what `wezterm.action.SelectTextAtMouseCursor 'SemanticZone'` provides but with copy feature for key bindings.

**Describe the solution you'd like**
Add the following methods to the copy mode

* `wezterm.action.CopyMode { SetSelectionMode = 'SemanticZone' }`
* `wezterm.action.CopyMode 'Move(Backward}Forward)SemanticZone'`
* `wezterm.action.CopyMode 'Move(Backward|Forward)Prompt'`

**Describe alternatives you've considered**

Use `wezterm.action.ScrollToPrompt(-1)` and `Pane.get_lines_as_text()` in combination with brute force operations in Lua native code.

**Additional context**

The proposal is the result from the discussion https://github.com/wez/wezterm/discussions/2343.

**Comment by wez** [comment]
In `main`, you can now use:

```lua
      -- move the cursor backwards to the start of the current zone, or
      -- to the prior zone if already at the start
      { key = 'z', mods = 'NONE', action = act.CopyMode 'MoveBackwardSemanticZone' },
      -- move the cursor forwards to the start of the next zone
      { key = 'Z', mods = 'NONE', action = act.CopyMode 'MoveForwardSemanticZone' },
      -- start selecting by zone: both the start point and the cursor
      -- position will be expanded to the containing zone and the union
      -- of those two will be used for the selection
      {
        key = 'z',
        mods = 'CTRL',
        action = act.CopyMode { SetSelectionMode = 'SemanticZone' },
      },
      -- like MoveBackwardSemanticZone by only considers zones of the
      -- specified type
      { key = 'z', mods = 'ALT', action = act.CopyMode { MoveBackwardZoneOfType ='Output' }},
      -- like MoveForwardSemanticZone by only considers zones of the
      -- specified type
      { key = 'Z', mods = 'ALT', action = act.CopyMode { MoveForwardZoneOfType ='Output' }},
```

It typically takes about an hour before fixes are available as nightly builds for all platforms.  Linux builds are the fastest to build and are often available within about 20 minutes.  Windows and macOS builds take a bit longer.

Please take a few moments to try out the fix and let me know how that works out. You can find the nightly downloads for your system in the [wezterm installation docs](https://wezfurlong.org/wezterm/installation.html).

If you prefer to use packages provided by your distribution or package manager of choice and don't want to replace that with a nightly download, keep in mind that you can download portable packages (eg: a `.dmg` file on macOS, a `.zip` file on Windows and an `.AppImage` file on Linux) that can be run without permanently installing or replacing an existing package, and can then simply be deleted once you no longer need them. 

If you are eager and can [build from source](https://wezfurlong.org/wezterm/install/source.html) then you may be able to try this out more quickly.

**Comment by ymotongpoo** [comment]
@wez Unbelievable!! This is absolutely fantastic and I updated my WezTerm to nightly using your homebrew cask! I confirmed new features worked perfectly and they made my day!

**Comment by wez** [comment]
Glad to hear that it is working for you!
I'll keep this open to remind myself to write up the docs!

**Comment by Doltonius** [comment]
I tried the mappings; this one, `{ key = 'z', mods = 'ALT', action = act.CopyMode { MoveBackwardZoneOfType ='Output' }},`, when issued at the top of the scrollback (so there is no more output zone above), brings wezterm to a halt, consuming 100% CPU. Should I open a new issue for this?

**Comment by wez** [comment]
@Doltonius thanks for reporting that; should be fixed now in 429c18b7e
